src/persist-client/src/write.rs

```diff
@@ -393,15 +430,20 @@ where
     /// Test helper for an [Self::append] call that is expected to succeed.
     #[cfg(test)]
     #[track_caller]
-    pub async fn expect_append(&mut self, updates: &[((K, V), T, D)], new_upper: T) {
+    pub async fn expect_append<L, U>(&mut self, updates: &[((K, V), T, D)], lower: L, new_upper: U)
+    where
+        L: Into<Antichain<T>>,
```

Comment:
I could change `append()` to use this as well, to make tests look slightly nicer? Maybe in a follow-up PR.

Reply:
hmm I can why you did this (so you can use the write handle's upper for lower), but it makes me a little sad to lose the simplicity of the single timestamp we had before. I don't really have anything concrete to counter-propose though...